app/src/main/java/org/wikipedia/readinglist/AddToReadingListDialog.java

```diff
@@ -220,29 +243,14 @@ private void addAndDismiss(final ReadingList readingList, final PageTitle title)
                     } else {
                         message = getString(R.string.reading_list_article_added_to_named, title.getDisplayText(), readingList.title());
                         new ReadingListsFunnel(title.getWikiSite()).logAddToList(readingList, readingLists.size(), invokeSource);
-
                         ReadingListDbHelper.instance().addPageToList(readingList, title, true);
                         showViewListSnackBar(readingList, message);
-
                     }
                     dismiss();
                 }, L::w));
     }
 
-    private void addAndDismiss(final ReadingList readingList, final List<PageTitle> titles) {
-
-        if ((readingList.pages().size() + titles.size()) > SiteInfoClient.getMaxPagesPerReadingList()) {
-            String message = getString(R.string.reading_list_article_limit_message, readingList.title(), SiteInfoClient.getMaxPagesPerReadingList());
-            FeedbackUtil.makeSnackbar(getActivity(), message, FeedbackUtil.LENGTH_DEFAULT).show();
-            dismiss();
-            return;
-        }
-
-        if (titles.size() == 1) {
-            addAndDismiss(readingList, titles.get(0));
-            return;
-        }
-
+    void execute(final ReadingList readingList, final List<PageTitle> titles) {
```

Comment:
Similarly, can we have just a single `execute` function that takes a list of Titles, and give it a singleton list in the case of a single title?

Reply:
Yes!